capytaine/io/legacy.py

```diff
@@ -259,3 +259,41 @@ def write_dataset_as_tecplot_files(results_directory, data):
                         fi.write('  ')
                     fi.write('\n')
 
+def export_hydrostatics(hydrostatics_directory, bodies):
+    """Determine filenames (following Nemoh convention) and call the .dat file writer"""
+
+    if os.path.isdir(hydrostatics_directory):
+        LOG.warning(f"""Exporting problem in already existing directory: {hydrostatics_directory}
+             You might be overwriting existing files!""")
+    else:
+        os.makedirs(hydrostatics_directory)
+
+    def hydrostatics_writer(hydrostatics_file_path, kh_file_path, body):
+        """Write the Hydrostatics.dat and KH.dat files"""
+        with open(hydrostatics_file_path, 'w') as hf:
+            for j in range(3):
+                line =  f'XF = {body.center_of_buoyancy[j]:7.3f} - XG = {body.center_of_mass[j]:7.3f} \n'
+                hf.write(line)
+            line = f'Displacement = {body.volume:E}'
+            hf.write(line)
+            hf.close()
+        np.savetxt(kh_file_path, body.hydrostatic_stiffness.values)
+
+    hydrostatics_file_name = "Hydrostatics.dat"
+    kh_file_name = "KH.dat"
+    try:
```

Comment:
Alternatively, you could do:
```python
if isinstance(bodies, FloatingBody) or len(bodies) == 1:
  ...  # Write single body
else:
  ...  # Write multiple body
```

Reply:
I did try that but you still end up having to do either `bodies.center_of_buoyancy` or `bodies[0].center_of_buoyancy`?

Reply:
Good point.

Otherwise, wrap the single `FloatingBody` in a list at the beginning of the function:
```python
def export_hydrostatics(hydrostatics_directory, bodies):
  if isinstance(bodies, FloatingBody):
    bodies = [bodies]
  ... # continue as if the input was a list
```

Because the `except TypeError` does not make it very explicit that this case is for a single body outside of a list.

Reply:
yeah thats a fair point - I've changed it to do FloatingBody -> [FloatingBody] 👍